### PR TITLE
i18n(zh-cn): Translate from-nextjs.mdx

### PR DESCRIPTION
**Pull request by qlly0630**
#### What kind of changes does this PR include?
- New or updated content
#### Description
- update src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx

### CODE REVIEW

**Comment by Yan-Thomas** [comment]
Thanks for the PR @qlly0630, I'm going to share your PR with other Chinese translators in our Discord, and would love to have you there so you can share any contributions you make as well! https://astro.build/chat

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx [hunk 1]
```diff
@@ -0,0 +1,574 @@
...
+	</head>
+	<body>
+		<slot/>
+	</body>
+</html>
+```
+
+### 迁移Pages和Posts
+
+在Next.js中，你的posts要么在`/pages`，要么在`/app/routeName/page.jsx`。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx:302]:
```suggestion
在Next.js中，你的文章要么在`/pages`，要么在`/app/routeName/page.jsx`。
```

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx [hunk 2]
```diff
@@ -0,0 +1,574 @@
...
+
+在Astro中，**你所有的页面内容必须在`src/`**中，在`src/pages`或`src/content`其中。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx:304]: Incorrect Markdown formatting here.

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx [hunk 3]
```diff
@@ -0,0 +1,574 @@
...
+
+#### React页面
+
+现有的Next JSX (.js)页面将需要从JSX文件转换为.astro页面。你不能在Astro中使用现有的JSX页面文件。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx:308]: Missing link!

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx [hunk 4]
```diff
@@ -0,0 +1,574 @@
...
+
+这些.astro页面必须位于src/pages/内，并将根据其文件路径自动生成页面路由。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx:310]: Missing link!

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx [hunk 5]
```diff
@@ -0,0 +1,574 @@
...
+
+#### Markdown和MDX页面
+
+Astro内置了对Markdown的支持和对MDX文件的可选整合。你可以复用任何现有的[Markdown和MDX文件](/zh-cn/guides/markdown-content/)，但它们可能需要对它们的frontmatter进行一些调整，比如添加[Astro的特殊`layout`前言属性](/zh-cn/corecepts/layouts/#markdownmdx-layouts)。你将不再需要为每个Markdown生成的路由手动创建页面。这些文件可以放在`src/pages/`中，利用文件即路由的优势自动生成。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx:314]:
```suggestion
Astro内置了对Markdown的支持和对MDX文件的可选整合。你可以复用任何现有的[Markdown和MDX文件](/zh-cn/guides/markdown-content/)，但它们可能需要对它们的frontmatter进行一些调整，比如添加[Astro的特殊`layout` frontmatter 属性](/zh-cn/core-concepts/layouts/#markdownmdx-布局)。你将不再需要为每个Markdown生成的路由手动创建页面。这些文件可以放在`src/pages/`中，利用文件即路由的优势自动生成。
```

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx [hunk 6]
```diff
@@ -0,0 +1,574 @@
...
+
+### 迁移测试
+
+由于Astro输出的是原始HTML，所以可以使用构建步骤的输出来编写端到端的测试。如果你已经能够匹配你的Next网站的标记，以前写的任何端到端测试都可能开箱即用。测试库，如Jest和React测试库，可以被导入并用于Astro测试你的React组件。
+
+参见Astro的[测试指南](/zh-cn/guides/testing/)了解更多。
+
+## 参考: 将Next.js语法转换为Astro语法
+
+### Links
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx:326]:
```suggestion
### 链接
```

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx [hunk 7]
```diff
@@ -0,0 +1,574 @@
...
+
+将任何Next `<Link to="">`, `<NavLink>`等组件转换成HTML `<a href="">`标签。
+
+```astro del={1} ins={2}
+<Link to="/blog">Blog</Link>
+<a href="/blog">Blog</a>
+```
+
+Astro没有为links使用任何特殊的组件，但我们欢迎你建立你自己的`<Link>`组件。然后你可以像使用其他组件一样导入和使用这个`<Link>`。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx:335]:
```suggestion
Astro没有为链接使用任何特殊的组件，但我们欢迎你建立你自己的`<Link>`组件。然后你可以像使用其他组件一样导入和使用这个`<Link>`。
```

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx [hunk 8]
```diff
@@ -0,0 +1,574 @@
...
+
+```astro title="src/components/Link.astro"
+---
+const { to } = Astro.props
+---
+<a href={to}><slot /></a>
+```
+
+### Imports
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx:344]:
```suggestion
### 导入
```

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx [hunk 9]
```diff
@@ -0,0 +1,574 @@
...
+
+更新任何[文件导入](/zh-cn/guides/imports/)以准确引用相对文件路径。这可以通过[import aliases](/zh-cn/guides/typescript/#import-aliases)来完成，或者通过写出相对路径的全称。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx:346]:
```suggestion
更新任何[文件导入](/zh-cn/guides/imports/)以准确引用相对文件路径。这可以通过[import aliases](/zh-cn/guides/typescript/#导入别名)来完成，或者通过写出相对路径的全称。
```

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx [hunk 10]
```diff
@@ -0,0 +1,574 @@
...
+      </div>
+    );  
+}
+
+<div>
+  <slot />
+</div>
+```
+
+可以使用[命名插槽](/zh-cn/corecepts/astro-components/#named-slots)将有多个子组件的React组件迁移到Astro组件。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx:377]:
```suggestion
可以使用[命名插槽](/zh-cn/core-concepts/astro-components/#命名插槽)将有多个子组件的React组件迁移到Astro组件。
```

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx [hunk 11]
```diff
@@ -0,0 +1,574 @@
...
+
+参考更多关于[Astro中具体的`<slot />`用法](/zh-cn/core-concepts/astro-components/#slots)。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx:379]:
```suggestion
参考更多关于[Astro中具体的`<slot />`用法](/zh-cn/core-concepts/astro-components/#插槽)。
```

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx [hunk 12]
```diff
@@ -0,0 +1,574 @@
...
+请注意，".astro "和其他一些文件类型必须以其完整的文件扩展名导入。
+
+```astro title="src/pages/authors/Fred.astro"
+---
+import Card from `../../components/Card.astro`
+---
+<Card />
+```
+
+### Children Props
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx:357]:
```suggestion
### 转换 Next 的 Children Props 到 Astro
```

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx [hunk 13]
```diff
@@ -0,0 +1,574 @@
...
+<div>
+  <slot />
+</div>
+```
+
+可以使用[命名插槽](/zh-cn/corecepts/astro-components/#named-slots)将有多个子组件的React组件迁移到Astro组件。
+
+参考更多关于[Astro中具体的`<slot />`用法](/zh-cn/core-concepts/astro-components/#slots)。
+
+### Fetching
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx:381]:
```suggestion
### 转换 Next 的数据获取到 Astro
```

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx [hunk 14]
```diff
@@ -0,0 +1,574 @@
...
+
+const response = await fetch('https://randomuser.me/api/');
+const data = await response.json();
+const randomUser = data.results[0];
+---
+```
+
+查看更多关于[用`Astro.glob()`导入本地文件](/zh-cn/guides/imports/#astroglob)，[使用Collections API查询](/zh-cn/guides/content-collections/#querying-collections)或[获取远程数据](/zh-cn/guides/Data-fetching/)。
+
+### Styling
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx:405]:
```suggestion
### 转换 Next 的样式到 Astro
```

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx [hunk 15]
```diff
@@ -0,0 +1,574 @@
...
+```astro title="src/components/Card.astro" del={1} ins={2}
+<div style={{backgroundColor: `#f4f4f4`, padding: `1em`}}>{message}</div>
+<div style="background-color: #f4f4f4; padding: 1em;">{message}</div>
+```
+
+在安装了[Tailwind集成](/zh-cn/guides/integrations-guide/tailwind/)之后，就支持Tailwind。不需要对您现有的Tailwind代码进行任何修改
+
+查看更多关于 [Styling in Astro](/zh-cn/guides/styling/)的信息。
+
+### Image
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx:420]:
```suggestion
### 转换 Next 的 图像插件到 Astro
```

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx [hunk 16]
```diff
@@ -0,0 +1,574 @@
...
+如果有必要，将任何内联样式对象（`style={{ fontWeight: "bold" }}`）转换成内联HTML样式属性（`style="font-weight:bold;"`）。或者，使用[Astro `<style>`标签](/zh-cn/guides/styling/#styling-in-astro)实现scoped CSS styles。
+
+```astro title="src/components/Card.astro" del={1} ins={2}
+<div style={{backgroundColor: `#f4f4f4`, padding: `1em`}}>{message}</div>
+<div style="background-color: #f4f4f4; padding: 1em;">{message}</div>
+```
+
+在安装了[Tailwind集成](/zh-cn/guides/integrations-guide/tailwind/)之后，就支持Tailwind。不需要对您现有的Tailwind代码进行任何修改
+
+查看更多关于 [Styling in Astro](/zh-cn/guides/styling/)的信息。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx:418]:
```suggestion
查看更多关于[在 Astro 中设计样式](/zh-cn/guides/styling/)的信息。
```

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx [hunk 17]
```diff
@@ -0,0 +1,574 @@
...
+---
+```
+
+查看更多关于[用`Astro.glob()`导入本地文件](/zh-cn/guides/imports/#astroglob)，[使用Collections API查询](/zh-cn/guides/content-collections/#querying-collections)或[获取远程数据](/zh-cn/guides/Data-fetching/)。
+
+### Styling
+
+你可能需要用Astro中其他可用的CSS选项来替换任何[CSS-in-JS库](https://github.com/withastro/astro/issues/4432)（例如styled-components）。
+
+如果有必要，将任何内联样式对象（`style={{ fontWeight: "bold" }}`）转换成内联HTML样式属性（`style="font-weight:bold;"`）。或者，使用[Astro `<style>`标签](/zh-cn/guides/styling/#styling-in-astro)实现scoped CSS styles。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx:409]:
```suggestion
如果有必要，将任何内联样式对象（`style={{ fontWeight: "bold" }}`）转换成内联HTML样式属性（`style="font-weight:bold;"`）。或者，使用[Astro `<style>`标签](/zh-cn/guides/styling/#在-astro-进行设计)实现scoped CSS styles。
```

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx [hunk 18]
```diff
@@ -0,0 +1,574 @@
...
+<div style="background-color: #f4f4f4; padding: 1em;">{message}</div>
+```
+
+在安装了[Tailwind集成](/zh-cn/guides/integrations-guide/tailwind/)之后，就支持Tailwind。不需要对您现有的Tailwind代码进行任何修改
+
+查看更多关于 [Styling in Astro](/zh-cn/guides/styling/)的信息。
+
+### Image
+
+用[Astro自己的图片集成组件](/zh-cn/guides/images/#astros-image-integration)，或用标准的HTML`<img>`来转换任何Next`<Image />`组件。请看[组件属性的完整列表](/zh-cn/guides/integrations-guide/image/#usage)，可用于Astro的`<Image />`和`<Picture />`组件，注意有几个属性和Next不同。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx:422]:
```suggestion
用[Astro自己的图片集成组件](/zh-cn/guides/images/#astro-的图片集成)，或用标准的HTML`<img>`来转换任何Next`<Image />`组件。请看[组件属性的完整列表](/zh-cn/guides/integrations-guide/image/#usage)，可用于Astro的`<Image />`和`<Picture />`组件，注意有几个属性和Next不同。
```

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx [hunk 19]
```diff
@@ -0,0 +1,574 @@
...
+
+## 示例: 数据请求
+
+下面是一个Next.js Pokédex数据获取转换为Astro的例子。
+    
+`pages/index.js`使用[REST PokéAPI](https://pokeapi.co/)获取并显示前151个神奇宝贝的列表。
+
+下面是如何在`src/pages/index.astro`中重新创建，将`getStaticProps()`替换为`fetch()`。
+
+1. 定义return() JSX。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx:448]:
```suggestion
1. 定义要返回的 JSX。
```

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx [hunk 20]
```diff
@@ -0,0 +1,574 @@
...
+        props: {
+            pokemons,
+        },
+    }
+}
+```
+
+2. 创建`src/pages/index.astro`
+
+使用Next函数的返回值。将任何Next或React语法转换成Astro，包括改变任何[HTML全局属性]的大小写(https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes)。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx:496]: Incorrect Markdown formatting!

MDN links maybe can use `zh-CN`

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx [hunk 21]
```diff
@@ -0,0 +1,574 @@
...
+
+注意:
+
+- `.map` 就已经可以了
+
+- `className` 变成 `class`.
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx:502]:
```suggestion
- `className` 变成 `class`。
```

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx [hunk 22]
```diff
@@ -0,0 +1,574 @@
...
+
+- `<Link>` 变成 `<a>`.
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx:504]:
```suggestion
- `<Link>` 变成 `<a>`。
```

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx [hunk 23]
```diff
@@ -0,0 +1,574 @@
...
+              </a>
+          </li>
+      ))}
+  </ul>
+</Layout>
+```
+
+## 社区资源
+
+- Video: [NextJS to Astro: more control = faster sites](https://www.youtube.com/watch?v=PSzCtdM20Fc).
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx:567]: There also need to be translated.

**Review by liruifengv — CHANGES_REQUESTED** [review]

**Comment by liruifengv** [comment]
感谢你的贡献 @qlly0630 为了方便我直接说中文，我标记了一些错误出来：
- 检查所有中文的链接，#号之后的fragment
- 一些 core-concepts -> corecepts 的链接错误
- Markdown 格式渲染错误
- MDN 等外部链接，如果有对应中文站点，可以指向中文站点，例如 https://developer.mozilla.org/zh-CN/
- 更通顺的中文翻译
- 中文和英文单词之间可以有空格，如果你愿意处理的话

**Comment by Yan-Thomas** [comment]
Hey @qlly0630, pinging for freshness! Could you commit liruifengv's suggestions to your PR? After this I think we're good to merge it!

**Comment by Yan-Thomas** [comment]
This PR got stale, I'll be drafting it for now till you have the time to look through the suggested changes so we can merge it, thanks!

**Comment by qlly0630** [comment]
@liruifengv I've finished fixing all the issues in the code review, help with the code review again.

**Comment by liruifengv** [comment]
> @liruifengv I've finished fixing all the issues in the code review, help with the code review again.↳

Please resolve the conversations.

**Comment by qlly0630** [comment]
> > @liruifengv I've finished fixing all the issues in the code review, help with the code review again.↳
> 
> Please resolve the conversations.

Done.

**Review by liruifengv — APPROVED** [review]
LGTM! Thanks for your continued follow-up @qlly0630

**Comment by qlly0630** [comment]
> One thing to note, a stable version of the app route layout for next.js has been released. We should update it in the original English version at a later date.

ok

File: src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx [hunk 24]
```diff
@@ -0,0 +1,574 @@
...
+// 获取所有 `src/pages/posts/` 的入口
+const allPosts = await Astro.glob('../pages/posts/*.md');
+
+const response = await fetch('https://randomuser.me/api/');
+const data = await response.json();
+const randomUser = data.results[0];
+---
+```
+
+查看更多关于[用`Astro.glob()`导入本地文件](/zh-cn/guides/imports/#astroglob)，[使用 Collections API 查询](/zh-cn/guides/content-collections/#querying-collections)或[获取远程数据](/zh-cn/guides/Data-fetching/)。
```
Comment (Yan-Thomas) [inline src/content/docs/zh-cn/guides/migrate-to-astro/from-nextjs.mdx:403]: Fix broken link
```suggestion
查看更多关于[用`Astro.glob()`导入本地文件](/zh-cn/guides/imports/#astroglob)，[使用 Collections API 查询](/zh-cn/guides/content-collections/#querying-collections)或[获取远程数据](/zh-cn/guides/data-fetching/)。
```

**Review by Yan-Thomas — APPROVED** [review]
LGTM! Nice work peeps 💜

**Pull request merged**